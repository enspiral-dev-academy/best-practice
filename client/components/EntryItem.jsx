import React from "react";
import { Link } from "react-router-dom";

import { wrappedWith, entryContext } from "../wrappers";

export function EntryItem({ selectEntry, entryData }) {
  return (
    <li data-testid="entry">
      <Link
        to={`/entry/${entryData.id}`}
        onClick={() => selectEntry(entryData)}
      >
        {entryData.name}
      </Link>
    </li>
  );
}

export default wrappedWith(entryContext)(EntryItem);
