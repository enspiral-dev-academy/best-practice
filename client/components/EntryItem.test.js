import React from "react";
import "@testing-library/jest-dom";
import { screen, fireEvent } from "@testing-library/react";
import { renderWithRouter } from "../testing/utils";

import { EntryItem } from "./EntryItem";
import mockEntries from "../testing/mockEntries";

test("renders an entry item correctly", () => {
  const entryData = mockEntries[0];
  const { asFragment } = renderWithRouter(<EntryItem entryData={entryData} />);
  expect(asFragment()).toMatchSnapshot();
});

test("calls selectEntry on Link click", async () => {
  const selectEntry = jest.fn();
  const entryData = mockEntries[0];
  renderWithRouter(
    <EntryItem selectEntry={selectEntry} entryData={entryData} />
  );

  const entryLink = await screen.getByRole("link", { name: "mocked entry 1" });
  fireEvent.click(entryLink);

  expect(selectEntry).toHaveBeenCalled();
});
