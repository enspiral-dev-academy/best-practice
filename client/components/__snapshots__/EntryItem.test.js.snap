// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an entry item correctly 1`] = `
<DocumentFragment>
  <li
    data-testid="entry"
  >
    <a
      href="/entry/1"
    >
      mocked entry 1
    </a>
  </li>
</DocumentFragment>
`;
